GameDesign/Assets/Scripts/MonoBehaviours/CustomerMonoBehavior.cs

```diff
@@ -0,0 +1,75 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CustomerMonoBehavior : MonoBehaviour, Clickable
+{
+    [SerializeField] public int id;
+    [SerializeField] public float baseSpeed;
+    [SerializeField] public float currentSpeed;
+    [SerializeField] public int clickAmount = 0;
+    [SerializeField] public int clickTime;
+    [SerializeField] public bool wearsMask;
+    [SerializeField] public NPCMovementManager moveManager;
+    [SerializeField] public int pointValue;
+    [SerializeField] public bool isFrozen = false;
+    [SerializeField] public int frozenTimeCount = 0;
+
+    [SerializeField] public TaserManager taserManager = TaserManager.Instance;
+
+    void Start()
+    {
+        InvokeRepeating("UnFreeze", 1.0f, 1.0f);
+        currentSpeed = baseSpeed;
+    }
+
+    public void Click(ClickType clickType)
+    {
+        if(clickType == ClickType.LEFT_CLICK)
+        {
+            clickAmount++;
```

Comment:
I don't see that `clickAmount` is doing anything. Shouldn't the wearMask=true; and the trigger be done when we click it X amount of times, where X changes between character type?